src/Widgets/CalendarChooser.vala

```diff
@@ -123,7 +117,24 @@ public class Maya.View.Widgets.CalendarChooser : Gtk.Grid {
             // TODO Should active calendar be re-selected?
         });
 
-        // Populate list_box
+        // Parse registry list_sources and render list_box;
+        render_sources ();
+
+        // Re-render sources when a new source connects;
+        calmodel.connected.connect (() => render_sources ());
+    }
+
+    private void render_sources () {
```

Comment:
Extracted render code into a private method so it can be used both in the initial constructor and as an event handler.